modules/highgui/src/XWDFile.h

```diff
@@ -0,0 +1,130 @@
+/*
+
+Copyright 1985, 1986, 1998  The Open Group
```

Comment:
Source of XWDFile.h Xmd.h files:

RedHat/CentOS - xorg-x11-proto-devel
Debian/Ubuntu - x11proto-dev
conda install -c conda-forge xorg-x11-proto-devel-cos7-x86_64

If you use a standard compiler, then there are no problems.

When using a cross compiler, internal header files are used, e.g. <gcc-riscv64-8.4.0/sysroot/usr/include> which do not contain the XWDFile.h and Xmd.h files and we hevn't found a way to install them there.


If you specify an additional path to the local system include directory:
-I/usr/include
the compilation process will be aborted due to overlapping include directories.

Thus, we decided that it would be easier to create copies of these files in our sources.

Reply:
We should not bring standard Linux headers.

`-dev` packages availability should be tested through `try_compile()` with appropriate includes and necessary structures.

For cross-compilation case users should have target-specific `sysroot` directory if extra dependencies are used.

Reply:
These files are needed only for XVFB support feature, only one data structure is used from these headers. Adding build dependency on x11 packages seems a bit excessive in this case and goes against the purpose of this PR - implement dependency-free GUI backend for Linux. 

So I see following possible ways here:
1. Leave these headers as-is, maybe add separate license file and include it into installation set. Maybe strip all unnecessary content from them.
2. Make optional build dependency on these headers and enable XVFB support feature only when they are found.
3. Remove XVFB support completely.

@opencv-alalek , what do you think? Which option should we choose?

Reply:
"Optional build feature" for this backend.

Reply:
Performed the following modifications:
1. Deleted XWDFile.h and Xmd.h files 
2. Added WITH_FRAMEBUFFER_XVFB option

When using WITH FRAMEBUFFER, the dependency on X11 is removed by defines.
When trying to use XVFB mode, a warning is issued asking you to change the OpenCV compilation options.

3. Added test code compilation check using XWD to <modules\highgui\CMakeLists.txt>

I'm not sure that the check is done in OpenCV style